src/WebJobs.Extensions.DurableTask/WebJobsConnectionInfoProvider.cs

```diff
@@ -0,0 +1,40 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+
+using System;
+using Microsoft.Extensions.Configuration;
+
+namespace Microsoft.Azure.WebJobs.Extensions.DurableTask
+{
+    /// <summary>
+    /// Connection info provider which resolves connection information from the WebJobs context.
+    /// </summary>
+    internal class WebJobsConnectionInfoProvider : IConnectionInfoResolver
+    {
+#if !FUNCTIONS_V1
+        private readonly IConfiguration hostConfiguration;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="WebJobsConnectionInfoProvider"/> class.
+        /// </summary>
+        /// <param name="hostConfiguration">A <see cref="IConfiguration"/> object provided by the WebJobs host.</param>
+        public WebJobsConnectionInfoProvider(IConfiguration hostConfiguration)
+        {
+            this.hostConfiguration = hostConfiguration ?? throw new ArgumentNullException(nameof(hostConfiguration));
+        }
+
+        /// <inheritdoc />
+        public IConfigurationSection Resolve(string name)
+        {
+            return this.hostConfiguration.GetWebJobsConnectionSection(name);
```

Comment:
I don't think so as previously the durable functions extension used the `IConnectionStringResolver` to resolve its connection string (using the config utility methods to prepend `AzureWebJobs`). Since this is still related to the connection, I think we can avoid using the `INameResolver` still.